MozillaFoundation/foundation.mozilla.org
# Updated signup form on primary nav

# Description

- Use the new React components to render the newsletter signup form on primary nav
- Remove old code in join.jsx

Link to sample test page: https://foundation-s-11655-prim-azv0pu.herokuapp.com/
Related PRs/issues: #11655

# Special test instructions for this particular form

1. Visit the review app link
2. Click on the "newsletter" button on nav
3. Make sure it toggles on the signup form section
4. Type something in the input field without submitting the form
5. Click outside of the section (i.e., anywhere outside of the black background), see if the this action resets the form and toggles off the expanded section
